database/src/migrations/20240119000000_taxonomy_tables.ts

```diff
@@ -0,0 +1,102 @@
+import { Knex } from 'knex';
+
+/**
+ * Add tables:
+ * - taxon
+ *
+ * @export
+ * @param {Knex} knex
+ * @return {*}  {Promise<void>}
+ */
+
+export async function up(knex: Knex): Promise<void> {
+  await knex.raw(`--sql
+    ----------------------------------------------------------------------------------------
+    -- Create tables
+    ----------------------------------------------------------------------------------------
+    set search_path=biohub,public;
+
+    CREATE TABLE taxon (
+      taxon_id                integer             GENERATED ALWAYS AS IDENTITY (START WITH 1 INCREMENT BY 1),
+      itis_tsn                integer             NOT NULL,
+      bc_taxon_code           varchar(10),
+      scientific_name         varchar(50)         NOT NULL,
+      common_name             varchar(50),
```

Comment:
Are we confident in these varchar lengths? I would probably go a bit higher just to be safe.

(The actual length specified has no impact on performance/storage, because it only actually stores the length of the string. Its just a constraint on the length.)

Reply:
I can bump them up. 100?

Reply:
Yeah 100 seems more reasonable. 

Wikipedia apparently has a page for long taxonomic names, the longest beign 73 characters. No idea how accurate it is, but 100 seems like it would cover all cases then.

https://en.wikipedia.org/wiki/List_of_long_species_names

Reply:
Actually, if we search ITIS by name for 'llanfair­pwll­gwyn­gyll­go­gery­chwyrn­drobwll­llan­tysilio­gogo­gochensis' I wonder if it comes up and how long the full scientific name is

Reply:
Per discussion, ITIS uses `varchar(300)` for their `completename` column.